difftest/dpi_t1rocketemu/src/bus/disp.rs

```diff
@@ -0,0 +1,149 @@
+use super::ShadowDevice;
+use std::fs::File;
+use std::io::BufWriter;
+use std::path::PathBuf;
+
+/// Device memory layout:
+/// `0x000_0000 - 0x1FF_0000` addressable frame buffer memory
+/// `0x1FF_0000 - 0x200_0000` control registers
+/// `0x1FF_0000`: read as frame counter, write to flush frame buffer as png
+/// `0x1FF_0004`: output dimensions
+///     0x04-06 16bits: OUTPUT_WIDTH (currently fixed at 960)
+///     0x06-08 16bits: OUTPUT_HEIGHT (currently fixed at 720)
+/// TODO: configurable dimension and color depth support?
+/// TODO: behavior emulation closer to actual LCD display?
+pub(super) struct DisplayDevice {
+  vram: Vec<u8>,
+  frame_counter: i32,
+}
```

Comment:
It's better to document the pixel format and layout somewhere. IIRC, we use RGB24 format. The size of framebuffer is `ROW * COL * 3`. Each pixel is represented by 3 consecutive uint8, denoting R,G,B component respectively.